src/transformers/modeling_utils.py

```diff
@@ -735,16 +736,37 @@ def _load_state_dict_into_meta_model(
             # For backward compatibility with older versions of `accelerate`
             set_module_tensor_to_device(model, param_name, param_device, **set_module_kwargs)
         else:
-            if param.dtype == torch.int8 and param_name.replace("weight", "SCB") in state_dict.keys():
-                fp16_statistics = state_dict[param_name.replace("weight", "SCB")]
-            else:
-                fp16_statistics = None
+            assert is_quantized, "something broke above this line"
```

Comment:
same as above

Reply:
Hmm I am also confused about this statement, why would something broke at this point?

Reply:
it was done when I was implementing [this suggestion above](https://github.com/huggingface/transformers/pull/26037#discussion_r1321097411). 
I changed `elif` code to be closer to your original suggestion and reworked the lines below it.

Reply:
it appeared when I was working on [this comment.](https://github.com/huggingface/transformers/pull/26037#discussion_r1321097411)
changed it closer to your original suggestion + reworked code below this line a bit.